File: Core/src/main/java/com/github/ezauton/core/actuators/VelocityMotor.java
package com.github.ezauton.core.actuators;

/**
 * A motor which can be run at a certain velocity
 */
public interface VelocityMotor extends Motor {
    /**
     * Run the motor at a certain velocity
     *
     * @param targetVelocity The *rotational* speed to run the motor at (RPM)
     */
    @Deprecated
    void runVelocity(double targetVelocity);

    default void runAngularVelocity(double targetVelocity) {
        runVelocity(targetVelocity);
    };
}

File: Core/src/main/kotlin/com/github/ezauton/core/actuators/VelocityMotor.kt
package com.github.ezauton.core.actuators

import com.github.ezauton.conversion.LinearVelocity

/**
 * A motor which can be run at a certain velocity
 */
interface VelocityMotor : Motor {
  /**
   * Run the motor at a certain velocity
   *
   * @param targetVelocity The speed to run the motor at
   */
  fun runVelocity(targetVelocity: LinearVelocity)
}
